dataland-frontend/src/components/pages/ApiKeysPage.vue

```diff
@@ -0,0 +1,235 @@
+      <template #footer>
+        <PrimeButton label="CANCEL" @click="regenerateConfirmToggle" class="p-button-outlined text-sm" />
+        <PrimeButton
+          label="CONFIRM"
+          @click="
+            () => {
+              setActivePageState('create');
+              regenerateConfirmToggle();
+            }
+          "
+          class="text-sm"
+        />
+      </template>
+    </PrimeDialog>
+  </AuthenticationWrapper>
+</template>
+
+<script lang="ts">
+import AuthenticationWrapper from "@/components/wrapper/AuthenticationWrapper.vue";
+import { defineComponent, inject, ref } from "vue";
+import PrimeButton from "primevue/button";
+import TheHeader from "@/components/generics/TheHeader.vue";
+import TheContent from "@/components/generics/TheContent.vue";
+import MiddleCenterDiv from "@/components/wrapper/MiddleCenterDivWrapper.vue";
+import BackButton from "@/components/general/BackButton.vue";
+import ApiKeyCard from "@/components/general/ApiKeyCard.vue";
+import CreateApiKeyCard from "@/components/general/CreateApiKeyCard.vue";
+import MessageComponent from "@/components/messages/MessageComponent.vue";
+import PrimeDialog from "primevue/dialog";
+import PrimeTextarea from "primevue/textarea";
+import { ApiClientProvider } from "@/services/ApiClients";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import Keycloak from "keycloak-js";
+
+export default defineComponent({
+  name: "ApiKeysPage",
+  components: {
+    AuthenticationWrapper,
+    TheContent,
+    TheHeader,
+    MiddleCenterDiv,
+    BackButton,
+    PrimeButton,
+    PrimeDialog,
+    ApiKeyCard,
+    CreateApiKeyCard,
+    MessageComponent,
+    PrimeTextarea,
+  },
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+      newKeyHolderRef: ref(),
+    };
+  },
+  data() {
+    return {
+      pageState: "view",
+      thereIsApiKey: false,
+      waitingForData: true,
+      regenerateConfirmationVisible: false,
+      newKey: "",
+      expiryDate: 0,
+    };
+  },
+  computed: {
+    PageTitleState() {
+      if (this.pageState === "view") return "API";
+      if (this.pageState === "create") return "Create new API Key";
+      return "API";
+    },
+  },
+  props: {},
+  mounted() {
+    void this.getApiKeyMetaInfoForUser();
+  },
+  watch: {},
+  methods: {
+    setActivePageState(state: string) {
+      this.pageState = state;
+    },
+
+    async getApiKeyMetaInfoForUser() {
+      try {
+        const keycloakPromiseGetter = assertDefined(this.getKeycloakPromise);
+        const resolvedKeycloakPromise = await keycloakPromiseGetter();
+        const apiKeyManagerController = await new ApiClientProvider(
+          keycloakPromiseGetter()
+        ).getApiKeyManagerController();
+        const apiKeyMetaInfoForUser = await apiKeyManagerController.getApiKeyMetaInfoForUser(
+          resolvedKeycloakPromise.subject as unknown as string
```

Comment:
The backend should figure out for which user the request was done. In line with my other comment.
The request shouldn't contain the keycloak_user_id